compiler/resolution/preFold.cpp

```diff
@@ -267,9 +267,15 @@ static void setRecordCopyableFlags(AggregateType* at) {
       }
 
     } else {
-      // Try resolving a test init= to set the flags
-      const char* err = NULL;
-      FnSymbol* initEq = findCopyInitFn(at, err);
+      // TODO Jade:
+      // special case since while implicit reads of sync/single are not yet removed
+      // with their removal, the init= that throws a warning will be gone
+      FnSymbol* initEq = NULL;
+      if(!ts->hasFlag(FLAG_SYNC) && !ts->hasFlag(FLAG_SINGLE)) {
+        // Try resolving a test init= to set the flags
+        const char* err = NULL;
+        initEq = findCopyInitFn(at, err);
+      }
       if (initEq == NULL) {
         ts->addFlag(FLAG_TYPE_INIT_EQUAL_MISSING);
```

Comment:
We might want to rename this flag (or probably better yet, set a new flag) to better describe what is going on, since we're explicitly ignoring if there's an `init=` for those types

Reply:
This flag is unused outside of `setRecordCopyableFlags` so the change is actually pretty minimal. In changing this, I would probably also change `FLAG_TYPE_ASSIGN_MISSING` which is similarly defined and used (and is only used in the analogous `setRecordAssignableFlags`).

I would propose `FLAG_TYPE_HAS_NO_INIT_EQUAL` and `FLAG_TYPE_HAS_NO_ASSIGN`

Reply:
Or alternatively, I think both of these flags can be removed entirely since the functions should work without them and they aren't used elsewhere.

Reply:
What about `FLAG_TYPE_SHOULDNT_INIT_EQUAL` or `FLAG_TYPE_DONT_USE_INIT_EQUAL`?

Reply:
I'd be okay with removing them entirely

Reply:
I have removed them and am running paratest now.

Reply:
Be sure to note that in your PR message before merging!

Reply:
paratest passed with everything removed, so I think this PR should be good to go